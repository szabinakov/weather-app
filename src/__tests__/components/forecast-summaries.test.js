import React from "react"
import { render } from "@testing-library/react"
import ForecastSummaries from '../../components/forecast-summaries'

describe('ForecastSummaries', () =>{
        const forecasts = [
            {
                date: 123,
                description: 'date1',
                icon: '200',
                temperature: {
                    max: 1234
                }
            },
            {
                date: 321,
                description: 'date2',
                icon: '900',
                temperature: {
                    max: 4321
                }
            },
            {
                date: 555,
                description: 'date3',
                icon: '800',
                temperature: {
                    max: 909
                }
            }
        ]
    it('it matches the snapshot', () =>{
        const {asFragment} = render(<ForecastSummaries forecasts={forecasts}/>)
        
        expect(asFragment).toMatchSnapshot()
    })
    it('renders the correct amount of props for ForecastSummaries', () => {
        const { getAllByTestId } = render ( <ForecastSummaries forecasts={forecasts}/> )

        expect(getAllByTestId('date-id')).toHaveLength(3)
        expect(getAllByTestId('description-id')).toHaveLength(3)
        expect(getAllByTestId('temperature-id')).toHaveLength(3)
        expect(getAllByTestId('icon-id')).toHaveLength(3)
    })
})