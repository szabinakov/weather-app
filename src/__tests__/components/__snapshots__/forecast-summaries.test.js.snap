// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries renders the correct amount of props for ForecastSummaries 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="date"
      data-testid="date-id"
    >
      123
    </div>
    <div
      class="temperature"
      data-testid="temperature-id"
    >
      1234
    </div>
    <div
      class="description"
      data-testid="description-id"
    >
      date1
    </div>
    <div
      class="icon"
      data-testid="icon-id"
    >
      icon0
    </div>
    <div
      class="date"
      data-testid="date-id"
    >
      321
    </div>
    <div
      class="temperature"
      data-testid="temperature-id"
    >
      4321
    </div>
    <div
      class="description"
      data-testid="description-id"
    >
      date2
    </div>
    <div
      class="icon"
      data-testid="icon-id"
    >
      icon1
    </div>
    <div
      class="date"
      data-testid="date-id"
    >
      555
    </div>
    <div
      class="temperature"
      data-testid="temperature-id"
    >
      909
    </div>
    <div
      class="description"
      data-testid="description-id"
    >
      date3
    </div>
    <div
      class="icon"
      data-testid="icon-id"
    >
      icon3
    </div>
  </div>
</DocumentFragment>
`;
